tests/WE2E/run_WE2E_tests.py

```diff
@@ -162,32 +155,37 @@ def run_we2e_tests(homedir, args) -> None:
         starttime_string = starttime.strftime("%Y%m%d%H%M%S")
         test_name = os.path.basename(test).split('.')[1]
         logging.debug(f"For test {test_name}, constructing config.yaml")
-        test_cfg = load_config_file(test)
-
-        if test_cfg.get('user') is None:
-            test_cfg['user'] = {}
-        test_cfg['user'].update({"MACHINE": machine})
-        test_cfg['user'].update({"ACCOUNT": args.account})
+        test_cfg = get_yaml_config(test)
+
+        test_config_updates = {
+             "user": {
+                "MACHINE": machine,
+                "ACCOUNT": args.account,
+                 },
+             "platform": {
+                 "BUILD_MOD_FN": args.modulefile,
+                 },
+             "workflow": {
+                 "COMPILER": args.compiler,
+                 "EXPT_SUBDIR": test_name
+                 },
+             }
         if run_envir:
-            test_cfg['user'].update({"RUN_ENVIR": run_envir})
-        # if platform section was not in input config, initialize as empty dict
-        if 'platform' not in test_cfg:
-            test_cfg['platform'] = dict()
-        test_cfg['platform'].update({"BUILD_MOD_FN": args.modulefile})
-        test_cfg['workflow'].update({"COMPILER": args.compiler})
+            test_config_updates['user'].update({"RUN_ENVIR": run_envir})
         if args.expt_basedir:
-            test_cfg['workflow'].update({"EXPT_BASEDIR": args.expt_basedir})
-        test_cfg['workflow'].update({"EXPT_SUBDIR": test_name})
+            test_config_updates['workflow'].update({"EXPT_BASEDIR": args.expt_basedir})
         if args.exec_subdir:
-            test_cfg['workflow'].update({"EXEC_SUBDIR": args.exec_subdir})
+            test_config_updates['workflow'].update({"EXEC_SUBDIR": args.exec_subdir})
         if args.launch == "cron":
-            test_cfg['workflow'].update({"USE_CRON_TO_RELAUNCH": True})
+            test_config_updates['workflow'].update({"USE_CRON_TO_RELAUNCH": True})
         if args.cron_relaunch_intvl_mnts:
-            test_cfg['workflow'].update({"CRON_RELAUNCH_INTVL_MNTS": args.cron_relaunch_intvl_mnts})
+            test_config_updates['workflow'].update({"CRON_RELAUNCH_INTVL_MNTS": args.cron_relaunch_intvl_mnts})
         if args.debug_tests:
-            test_cfg['workflow'].update({"DEBUG": args.debug_tests})
+            test_config_updates['workflow'].update({"DEBUG": args.debug_tests})
         if args.verbose_tests:
-            test_cfg['workflow'].update({"VERBOSE": args.verbose_tests})
+            test_config_updates['workflow'].update({"VERBOSE": args.verbose_tests})
+
+        test_cfg.update_from(test_config_updates)
```

Comment:
In general, I find this much easier to follow since some of the original lines _always_ got set, while others depended on command line args.